homeassistant/components/incomfort/climate.py

```diff
@@ -26,18 +27,20 @@
 class InComfortClimate(ClimateDevice):
     """Representation of an InComfort/InTouch climate device."""
 
-    def __init__(self, client, room):
+    def __init__(self, client, room) -> None:
         """Initialize the climate device."""
+        self.entity_id = f"climate.{DOMAIN}_{room.room_no}"
```

Comment:
Why do we need to set entity_id explicitly?

Reply:
The old names were:
 - `water_heater.boiler`, and
 - `climate.room_1`, `climate.room_2`

These were seen to be not unique across the platform (one person complained to me about the climate entity), so the need to change the name.  

I took the opportunity to have a descriptive `name` (long), and a (likely) unique `entity_id` (short).
 - `water_heater.incomfort`
 - `climate.incomfort_1`, `climate.incomfort_2`

Reply:
Ok. There's no possibility of setting a unique_id per entity?

Reply:
Yes, I could use the:
a) ordinal of the Heater for the Gateway (0, 1, 2...), perfect only if no-one has multiple Gateways, or
b) the Manufacturer's serial number of the Heater, e.g. "179t23082"

For example:
 - `water_heater.incomfort_0` or `water_heater.incomfort_179t23082`
 - `climate.incomfort_0_1`, or `climate.incomfort_179t23082_1`

I don't think anyone would have two Intergas Gateways, or even two Heaters on the one Gateway (although _I_ was very close to having multiple Gateways - with 1 Heater each, they would have been on different instances of HA).

The real issue was that the old names were very likely to clash:
 - `water_heater.boiler`, and
 - `climate.room_1`

Reply:
Is the serial number unique per gateway?

Reply:
Yes. In fact, serial number unique to all Heaters by that manufacturer - nothing to do with the Gateway, other than the Gateway is reporting it. 

@MartinHjelmare I also have #26640 - do you have time to look at it before .100 goes beta?

Reply:
Then we should instead do this:

1.For this release we implement unique_id entity property, but don't change name or set an entity_id.
2. For next release we change default name and set a new entity_id.

This will avoid a breaking change since we don't change anything until after the entity registry has been activated for users.

Reply:
OK, try that!

I added unique_id to all entities, based upon `f"{DOMAIN}_(SERIAL_NO}_(slugify(something)}"`.